### ISSUE TITLE
Failed deployments to sagemaker cannot be deleted from deployments list

### ISSUE THREAD

**Issue by omrihar** [bentoml/BentoML#504]
**Describe the bug**
I tried to deploy to sagemaker while debugging a problem with deployment to AWS Lambda. Twice the deployment failed, one because of a bug in the deployment script ("Service" tag does not appear for all AWS Roles, see https://bentoml.slack.com/archives/CKRANBHPH/p1577026534009000) and once because of some permissions issue with this role.
The problem is that now both failed deployments appear in the deployment list and delete fails because it does not find the endpoints.

**To Reproduce**
1. Make sure a deployment script to sagemaker will fail.
2. Deploy to sagemaker
3. Run `bentoml deployment delete <deployment name>`

**Expected behavior**
The deployment should be deleted as best is possible and not appear in the list.

**Screenshots/Logs**
```
[2020-01-29 09:28:38,670] ERROR - AWS ClientError - operation: DeleteEndpoint, code: ValidationException, message: Could not find endpoint "arn:aws:sagemaker:eu-central-1:436054152060:endpoint/dev-hektor-sagemaker-2".
Failed to delete deployment hektor-sagemaker-2. INTERNAL:AWS ClientError - operation: DeleteEndpoint, code: ValidationException, message: Could not find endpoint "arn:aws:sagemaker:eu-central-1:436054152060:endpoint/dev-hektor-sagemaker-2".

```

**Environment:**
 - OS: `Linux omri-pc 5.4.14-2-MANJARO #1 SMP PREEMPT Fri Jan 24 09:34:16 UTC 2020 x86_64 GNU/Linux`
 - Python/BentoML Version Python 3.7.6, BentoML-0.6.1

**Comment by parano** [comment]
Hi @omrihar , I believe the first issue you mentioned has been fixed in #510 

To delete a deployment record without confirming cloud resource being deleted, use the `--force` option: `bentoml deployment delete -f <deployment name>`

**Comment by parano** [comment]
PR #522 has also just introduced a fix for the second issue when the SageMaker endpoint has already been deleted.

**Comment by parano** [comment]
0.6.2 has just been released with the fix in #522  https://github.com/bentoml/BentoML/releases/tag/v0.6.2

@omrihar feel free to reopen if you ran into any issue with the new version
